src/cache_cf.cc

```diff
@@ -1018,87 +1021,138 @@ parse_obsolete(const char *name)
+/// returns the time as a chrono duration object of 'TimeUnit' type.
+/// \param defaultUnitName will be used if the time unit is missing,
+/// it should passed only if the configuration paramter being parsed
+/// excpects more arguments after time unit.
+template <class TimeUnit>
+static TimeUnit
+parseTimeLine(const char *defaultUnitName = nullptr)
 {
-    if (allowMsec && !strncasecmp(unit, T_MILLISECOND_STR, strlen(T_MILLISECOND_STR)))
-        return 1;
-
-    if (!strncasecmp(unit, T_SECOND_STR, strlen(T_SECOND_STR)))
-        return 1000;
+    const auto valueToken = ConfigParser::NextToken();
+    if (!valueToken) {
+        self_destruct();
+        return TimeUnit::zero();
+    }
 
-    if (!strncasecmp(unit, T_MINUTE_STR, strlen(T_MINUTE_STR)))
-        return 60 * 1000;
+    const auto parsedValue = xatof(valueToken);
 
-    if (!strncasecmp(unit, T_HOUR_STR, strlen(T_HOUR_STR)))
-        return 3600 * 1000;
+    std::chrono::nanoseconds parsedUnitDuration;
 
-    if (!strncasecmp(unit, T_DAY_STR, strlen(T_DAY_STR)))
-        return 86400 * 1000;
+    const auto token = parsedValue ? ConfigParser::PeekAtToken() : nullptr;
 
-    if (!strncasecmp(unit, T_WEEK_STR, strlen(T_WEEK_STR)))
-        return 86400 * 7 * 1000;
+    if (parsedValue && parseTimeUnits<TimeUnit>(token, defaultUnitName, parsedUnitDuration))
+        (void)ConfigParser::NextToken();
 
-    if (!strncasecmp(unit, T_FORTNIGHT_STR, strlen(T_FORTNIGHT_STR)))
-        return 86400 * 14 * 1000;
+    CheckTimeValue(parsedValue, parsedUnitDuration);
 
-    if (!strncasecmp(unit, T_MONTH_STR, strlen(T_MONTH_STR)))
-        return static_cast<uint64_t>(86400) * 30 * 1000;
+    const std::chrono::nanoseconds resultDuration(static_cast<std::chrono::nanoseconds::rep>(parsedUnitDuration.count() * parsedValue));
 
-    if (!strncasecmp(unit, T_YEAR_STR, strlen(T_YEAR_STR)))
-        return static_cast<uint64_t>(86400 * 1000 * 365.2522);
+    // validate precisions (time-units-small only)
+    if (TimeUnit(1) <= std::chrono::microseconds(1)) {
+        const auto durationNanoseconds = resultDuration.count();
+        if (durationNanoseconds && durationNanoseconds <= 3) {
+            const auto unitName = token ? token : defaultUnitName;
+            assert(unitName);
+            debugs(3, DBG_CRITICAL, "WARNING: The parsed '" << parsedValue << " " << unitName <<
+                    "' in '" << config_input_line << "' is too small to be measured with an acceptable precision");
+        }
+    }
 
-    if (!strncasecmp(unit, T_DECADE_STR, strlen(T_DECADE_STR)))
-        return static_cast<uint64_t>(86400 * 1000 * 365.2522 * 10);
+    const auto result = std::chrono::duration_cast<TimeUnit>(resultDuration);
 
-    debugs(3, DBG_IMPORTANT, "parseTimeUnits: unknown time unit '" << unit << "'");
+    if (parsedValue && !result.count()) {
+        debugs(3, DBG_CRITICAL, "FATAL: Invalid time value '" << parsedValue <<
+                "' is to small to be used in this context");
```

Comment:
```suggestion
                "' is too small to be used in this context");
```

Reply:
If it is easy, please print the smallest acceptable value as well. And check other, similar errors for similar problems/enhancements.